mlflow/tensorflow/__init__.py

```diff
@@ -887,14 +897,113 @@ def _patch_implementation(
                     # & tf.keras callbacks if necessary
                     callbacks = list(kwargs.get("callbacks") or [])
                     kwargs["callbacks"], self.log_dir = _setup_callbacks(
-                        callbacks, log_models, metrics_logger
+                        callbacks, False, metrics_logger
                     )
 
                 early_stop_callback = _get_early_stop_callback(callbacks)
                 _log_early_stop_callback_params(early_stop_callback)
 
                 history = original(inst, *args, **kwargs)
 
+                if log_models:
+                    import numpy as np
+
+                    def _is_x_type_supported_for_autologging(input_data):
```

Comment:
Do we need this function, or can we modify the behavior of `_get_input_data_slice` to return `None` if the input data is of an unsupported / unrecognized type for input example extraction?